astropy/coordinates/representation.py

```diff
@@ -435,6 +467,78 @@ class BaseRepresentation(BaseRepresentationOrDifferential):
 
     recommended_units = {}  # subclasses can override
 
+    def __init__(self, *args, **kwargs):
```

Comment:
More debatable here, but it would probably be useful for this to have a docstring of what's valid for ``differentials`` here